src/components/Card.js

```diff
@@ -1,50 +1,97 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
-    this._name = cardData.name;
-    this._link = cardData.link;
+  constructor({ name, link, likes, _id, userId }, myId, cardSelector, handleCardClick, handleDeleteClick, handleCardlike) {
+    this._name = name;
+    this._link = link;
+    this._likes = likes;
+    this._id = _id;
+    this._myId = myId;
+    this._userId = userId;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleCardLike = handleCardlike;
   }
 
   _getTemplate() {
-    this._cardElement = document.querySelector(this._cardSelector).content.querySelector(".card__list-item").cloneNode(true);
-    return this._cardElement;
+    return document.querySelector(this._cardSelector).content.querySelector(".card__list-item").cloneNode(true);
   }
 
-  //handleLikeButton
-  _toggleLikeButton() {
-    this._cardElement.querySelector(".card__like-button").classList.toggle("card__like-button_active");
+  _fillCardTemplate() {
+    const imageElement = this._element.querySelector(".card__image");
+    imageElement.alt = this._name;
+    imageElement.src = this._link;
+    this._element.id = this.imageID;
+    this._element.querySelector(".card__title").textContent = this._name;
   }
-  //handleDeleteButton
-  _deleteCard() {
-    this._cardElement.remove();
-    this._cardElement = null;
+
+  _getData() {
+    const data = {
+      name: this._name,
+      link: this._link,
+    };
+    return data;
+  }
+
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => {
+      return like._id === this._myId;
+    });
+  }
+
+  setLikes(likes) {
+    this._likes = likes;
+    this.renderLikes();
+  }
+
+  renderLikes() {
+    this._element.querySelector(".card__like-counter").textContent = this._likes.length;
```

Comment:
`this._element.querySelector(".card__like-counter")`



**Could be improved**




All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.